static/app/views/performance/landing/widgets/widgets/slowScreens.tsx

```diff
@@ -0,0 +1,396 @@
+import {
+  GrowLink,
+  WidgetEmptyStateWarning,
+} from 'sentry/views/performance/landing/widgets/components/selectableList';
+import {transformDiscoverToList} from 'sentry/views/performance/landing/widgets/transforms/transformDiscoverToList';
+import {transformEventsRequestToArea} from 'sentry/views/performance/landing/widgets/transforms/transformEventsToArea';
+import {
+  PerformanceWidgetProps,
+  QueryDefinition,
+  QueryDefinitionWithKey,
+  WidgetDataConstraint,
+  WidgetDataResult,
+  WidgetPropUnion,
+} from 'sentry/views/performance/landing/widgets/types';
+import {
+  eventsRequestQueryProps,
+  getMEPParamsIfApplicable,
+  QUERY_LIMIT_PARAM,
+  TOTAL_EXPANDABLE_ROWS_HEIGHT,
+} from 'sentry/views/performance/landing/widgets/utils';
+import {Subtitle} from 'sentry/views/profiling/landing/styles';
+import {RightAlignedCell} from 'sentry/views/replays/deadRageClick/deadRageSelectorCards';
+import Chart from 'sentry/views/starfish/components/chart';
+import {useReleaseSelection} from 'sentry/views/starfish/queries/useReleases';
+import {SpanMetricsField} from 'sentry/views/starfish/types';
+import {STARFISH_CHART_INTERVAL_FIDELITY} from 'sentry/views/starfish/utils/constants';
+import {RoutingContextProvider} from 'sentry/views/starfish/utils/routingContext';
+import {OUTPUT_TYPE, YAxis, YAXIS_COLUMNS} from 'sentry/views/starfish/views/screens';
+
+type DataType = {
+  chart: WidgetDataResult & ReturnType<typeof transformEventsRequestToArea>;
+  list: WidgetDataResult & ReturnType<typeof transformDiscoverToList>;
+};
+
+export function transformEventsChartRequest<T extends WidgetDataConstraint>(
+  widgetProps: WidgetPropUnion<T>,
+  results: RenderProps,
+  _: QueryDefinitionWithKey<T>
+) {
+  const {start, end, utc, interval, statsPeriod} = normalizeDateTimeParams(
+    widgetProps.location.query
+  );
+
+  const data = results.results ?? [];
+
+  const childData = {
+    ...results,
+    isLoading: results.loading || results.reloading,
+    isErrored: results.errored,
+    hasData: defined(data) && !!data.length && !!data[0].data.length,
+    data,
+    previousData: results.previousTimeseriesData ?? undefined,
+
+    utc: utc === 'true',
+    interval,
+    statsPeriod: statsPeriod ?? undefined,
+    start: start ?? '',
+    end: end ?? '',
+  };
+
+  return childData;
+}
+
+function SlowScreensByTTID(props: PerformanceWidgetProps) {
+  const api = useApi();
+  const pageFilter = usePageFilters();
+  const mepSetting = useMEPSettingContext();
+  const {
+    isLoading: isLoadingReleases,
+    primaryRelease,
+    secondaryRelease,
+  } = useReleaseSelection();
+  const location = useLocation();
+  const [selectedListIndex, setSelectListIndex] = useState<number>(0);
+  const {organization, InteractiveTitle} = props;
+  const pageError = usePageError();
+
+  const field = props.fields[0];
+
+  const listQuery = useMemo<QueryDefinition<DataType, WidgetDataResult>>(
```

Comment:
do you know if it's possible to not run this until the release selection is settled? it looks like we fire this request with release:[undefined,undefined] the first time

Reply:
Yeah, I fixed this by catching `isLoadingReleases` and returning null instead of the Discover query component, so the query only happens when the releases are loaded.

One thing I noticed, and tried to fix, was that deferring the query component means the loading state doesn't get triggered until the releases are done loading, so there's a quick flash of the empty state. We don't expose an `isLoading` override for the components further down, do you think this is a blocker that we should address?